yt/geometry/tests/test_particle_octree.py

```diff
@@ -398,6 +406,70 @@ def test_bitmap_select():
                     assert_array_equal(gf, gf_ans, f"selector {i}, ghost file array")
 
 
+@requires_file("TNGHalo/halo_59.hdf5")
+def test_ewah_write_load(tmp_path):
+    mock_file = tmp_path / "halo_59.hdf5"
+    mock_file.symlink_to(_get_test_data_dir_path() / "TNGHalo" / "halo_59.hdf5")
+
+    ds = load(mock_file)
+    _, c = ds.find_max(("gas", "density"))
+    assert ds.index.pii.mutable_index
+    assert ds.index.pii.order1 == 6
+    assert ds.index.pii.order2_orig == 2
+    assert ds.index.pii.order2 == 4
+    assert not ds.index.pii.loaded
+    c += ds.quan(0.5, "Mpc")
+    sp = ds.sphere(c, (20.0, "kpc"))
+    mass = sp.sum(("gas", "mass"))
+
+    del ds
+
+    ds2 = load(mock_file)
+    _, c2 = ds2.find_max(("gas", "density"))
+    assert ds2.index.pii.mutable_index
+    assert ds2.index.pii.order1 == 6
+    assert ds2.index.pii.order2_orig == 2
+    assert ds2.index.pii.order2 == 4
+    assert ds2.index.pii.loaded
+    c2 += ds2.quan(0.5, "Mpc")
+    sp2 = ds2.sphere(c, (20.0, "kpc"))
+    mass2 = sp2.sum(("gas", "mass"))
+
+    assert_array_equal(mass, mass2)
+
+    del ds2
+
+    ds3 = load(mock_file, index_order=(5, 3))
+    _, c3 = ds3.find_max(("gas", "density"))
+    assert not ds3.index.pii.mutable_index
+    assert ds3.index.pii.order1 == 5
+    assert ds3.index.pii.order2_orig == 3
+    assert ds3.index.pii.order2 == 3
+    assert not ds3.index.pii.loaded
+    c3 += ds3.quan(0.5, "Mpc")
+    sp3 = ds3.sphere(c, (20.0, "kpc"))
+    mass3 = sp3.sum(("gas", "mass"))
+
+    assert_array_equal(mass, mass3)
+
+    del ds3
+
+    ds4 = load(mock_file, index_order=(5, 3))
```

Comment:
if we're going for more than 2 repeats, I think 3 would suffice, and we might as well make it in an actual loop to ensure that various repeats are perfectly equivalent